src/realm/table.cpp

```diff
@@ -1493,44 +1443,41 @@ void Table::migrate_links(util::FunctionRef<void()> commit_and_continue)
             link_column_accessors[col_ndx]->init_from_parent();
             col_keys[col_ndx] = m_spec.get_key(col_ndx);
             col_types[col_ndx] = col_type;
-        }
-    }
-
-    size_t sz = size_t(-1);
-    for (auto& acc : link_column_accessors) {
-        if (acc) {
-            if (sz == size_t(-1)) {
-                sz = acc->size();
-            }
-            else {
-                REALM_ASSERT(acc->size() == sz);
-            }
-        }
-    }
-    if (sz != size_t(-1)) {
-        bool use_oid = (m_spec.get_column_name(0) == "!OID");
-        BPlusTree<Int> oid_column(m_alloc);
-        if (use_oid) {
-            oid_column.init_from_ref(to_ref(col_refs.get(0)));
-        }
-        for (size_t i = 0; i < sz; i++) {
-            ObjKey obj_key(use_oid ? oid_column.get(i) : i);
-            Obj obj = get_object(obj_key);
-            for (size_t col_ndx = 0; col_ndx < nb_columns; col_ndx++) {
-                if (col_keys[col_ndx]) {
-                    int64_t val = link_column_accessors[col_ndx]->get(i);
-                    if (val) {
-                        if (col_types[col_ndx] == col_type_Link) {
-                            obj.set(col_keys[col_ndx], ObjKey(val - 1));
-                        }
-                        else {
-                            auto ll = obj.get_linklist(col_keys[col_ndx]);
-                            BPlusTree<Int> links(m_alloc);
-                            links.init_from_ref(ref_type(val));
-                            size_t nb_links = links.size();
-                            for (size_t j = 0; j < nb_links; j++) {
-                                ll.add(ObjKey(links.get(j)));
-                            }
+            target_tables[col_ndx] = get_opposite_table(col_keys[col_ndx]).unchecked_ptr();
+            opposite_orig_row_ndx_col[col_ndx] = target_tables[col_ndx]->get_column_key("!ROW_INDEX");
+        }
+    }
+
+    auto orig_row_ndx_col = get_column_key("!ROW_INDEX");
+    for (auto obj : *this) {
+        for (size_t col_ndx = 0; col_ndx < nb_columns; col_ndx++) {
+            if (col_keys[col_ndx]) {
+                // If no !ROW_INDEX column is found, the original row index number is
```

Comment:
perhaps use a properly named boolean instead of testing orig_row_ndx_col ? (which by the way is a key, but includes ndx in its name??)

Reply:
Well `ndx` refers to `row`  - not the column. I think it makes sense to test `orig_row_ndx_col` before using it.